src/announce_entry.cpp

```diff
@@ -99,14 +99,15 @@ namespace libtorrent
 		min_announce = min_time();
 	}
 
-	void announce_entry::failed(aux::session_settings const& sett, int retry_interval)
+	void announce_entry::failed(time_duration const tracker_backoff, int const retry_interval)
 	{
 		++fails;
 		// the exponential back-off ends up being:
 		// 7, 15, 27, 45, 95, 127, 165, ... seconds
 		// with the default tracker_backoff of 250
+		int const tracker_backoff_seconds = total_seconds(tracker_backoff);
 		int delay = (std::min)(tracker_retry_delay_min + int(fails) * int(fails)
```

Comment:
did it turn out to be messy to maintain the `time_duration` type here? (rather than casting to int and then back again)